// html-export-pdf-cli/dist/constants/bin.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/constants/bin.ts
var bin_exports = {};
__export(bin_exports, {
  HTML_EXPORT_PDF_CLI: () => HTML_EXPORT_PDF_CLI
});
module.exports = __toCommonJS(bin_exports);

// package.json
var package_default = {
  name: "html-export-pdf-cli",
  type: "module",
  version: "1.2.0",
  packageManager: "pnpm@9.6.0",
  description: "Render HTML to PDF(with outline) using Puppeteer.",
  author: "CondorHero",
  license: "MIT",
  homepage: "https://github.com/condorheroblog/html-export-pdf-cli#readme",
  repository: {
    type: "git",
    url: "git+https://github.com/condorheroblog/html-export-pdf-cli.git"
  },
  bugs: {
    url: "https://github.com/condorheroblog/html-export-pdf-cli/issues"
  },
  keywords: [
    "html-export-pdf-cli",
    "html-export-outline-pdf",
    "html-export-bookmark-pdf",
    "html-export-pdf",
    "html-to-pdf-cli",
    "bookmark-pdf",
    "outline-pdf",
    "html-to-pdf",
    "htmltopdf",
    "html2pdf",
    "pdf-cli"
  ],
  exports: {
    ".": {
      import: "./dist/index.js",
      require: "./dist/index.cjs"
    }
  },
  main: "dist/index.cjs",
  module: "dist/index.js",
  types: "dist/index.d.ts",
  typesVersions: {
    "*": {
      "*": [
        "./dist/*",
        "./dist/index.d.ts"
      ]
    }
  },
  bin: {
    "html-export-pdf-cli": "bin/html-export-pdf-cli.mjs"
  },
  files: [
    "bin",
    "dist"
  ],
  engines: {
    node: ">=18.12.0"
  },
  scripts: {
    prepublishOnly: "nr build",
    lint: "eslint .",
    "lint:fix": "eslint . --fix",
    dev: "jiti ./src/index.ts",
    coverage: "vitest run --coverage",
    test: "vitest",
    build: "tsup",
    "build:watch": "tsup --watch",
    typecheck: "tsc --noEmit",
    release: "bumpp -r && npm publish"
  },
  workspaces: [
    "examples/*"
  ],
  dependencies: {
    "cli-progress": "^3.12.0",
    commander: "^12.1.0",
    "fast-glob": "^3.3.2",
    "html-entities": "^2.5.2",
    "pdf-lib": "1.17.1",
    puppeteer: "^22.15.0"
  },
  devDependencies: {
    "@antfu/eslint-config": "^2.24.1",
    "@types/cli-progress": "^3.11.6",
    "@types/node": "^22.0.2",
    bumpp: "^9.4.1",
    colorette: "^2.0.20",
    eslint: "^9.8.0",
    jiti: "^1.21.6",
    "lint-staged": "^15.2.7",
    "simple-git-hooks": "^2.11.1",
    tsup: "^8.2.3",
    typescript: "^5.5.4",
    vitest: "^2.0.5"
  },
  "simple-git-hooks": {
    "pre-commit": "npx lint-staged"
  },
  "lint-staged": {
    "*": "eslint --fix"
  },
  publishConfig: {
    access: "public",
    registry: "https://registry.npmjs.org"
  }
};

// src/constants/bin.ts
var HTML_EXPORT_PDF_CLI = package_default.name;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  HTML_EXPORT_PDF_CLI
});
